tests/test_sensor.py

```diff
@@ -80,6 +82,49 @@ def status_func(value):
     assert sensor.status == Sensor.Status.NOMINAL
 
 
+def test_sensor_value_setter(diverse_sensors):
+    """Check whether the value type verification works."""
+    diverse_sensors["test-bool-sensor"].value = True
+    assert diverse_sensors["test-bool-sensor"]
+    with pytest.raises(TypeError):
+        diverse_sensors["test-bool-sensor"].value = "True"
+
+    diverse_sensors["test-int-sensor"].value = 1234
+    assert diverse_sensors["test-int-sensor"].value == 1234
+    with pytest.raises(TypeError):
+        diverse_sensors["test-int-sensor"].value = "1234"
+
+    diverse_sensors["test-float-sensor"].value = 1234.5
+    assert diverse_sensors["test-float-sensor"].value == 1234.5
+    with pytest.raises(TypeError):
+        diverse_sensors["test-float-sensor"].value = "1234"
+
+    diverse_sensors["test-str-sensor"].value = "foo"
+    assert diverse_sensors["test-str-sensor"].value == "foo"
+    with pytest.raises(TypeError):
+        diverse_sensors["test-str-sensor"].value = 1234
+
+    diverse_sensors["test-bytes-sensor"].value = b"bar"
+    assert diverse_sensors["test-bytes-sensor"].value == b"bar"
+    with pytest.raises(TypeError):
+        diverse_sensors["test-bytes-sensor"].value = "foo"
+
+    diverse_sensors["test-address-sensor"].value = "1.2.3.4"
+    assert diverse_sensors["test-address-sensor"].value == Address("1.2.3.4")
```

Comment:
That's not a valid Address. The first parameter to Address must be IPv4Address | IPv6Address. It may be worth adding a runtime check to the Address constructor to catch places that people are messing this up.